receiver/chronyreceiver/internal/chrony/types.go

```diff
@@ -0,0 +1,176 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chrony // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/chronyreceiver/internal/chrony"
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"math"
+	"net"
+	"time"
+
+	"github.com/facebook/time/ntp/chrony"
+)
+
+/*
+	Notice:
+
+	This file's contents has been taken and modified from:
```

Comment:
Then you need to keep the copyright headers. Just add the original ones to the top of this source file, keeping the OpenTelemetry ones, as you made changes as well.